### PR TITLE
add tqdm progress bar to embedding creation step.

### LINKED ISSUE #42: Add progress bar for spec2vec.matrix() calculation
Most time consuming is the generation of the embeddings it seems. So we could add a ``tqdm`` progress bar to that.

### PR DESCRIPTION
**Pull request by florian-huber**
- fix conda workflow (analog to matchms)
- add ``tqdm`` progress bar to embedding calculation in Spec2Vec (optional, default is off)

### CODE REVIEW

File: spec2vec/Spec2Vec.py [hunk 1]
```diff
@@ -109,14 +117,14 @@ def matrix(self, references: List[SpectrumDocument], queries: List[SpectrumDocum
         """
         n_rows = len(references)
         reference_vectors = numpy.empty((n_rows, self.vector_size), dtype="float")
-        for index_reference, reference in enumerate(references):
+        for index_reference, reference in enumerate(tqdm(references, disable=self.disable_progress_bar)):
```
Comment (sverhoeven) [inline spec2vec/Spec2Vec.py]: Would be nice to see in stdout what the progress is for
```suggestion
        for index_reference, reference in enumerate(tqdm(references, desc='Calculating vectors of reference spectrums', disable=self.disable_progress_bar)):
```

File: spec2vec/Spec2Vec.py [hunk 2]
```diff
@@ -109,14 +117,14 @@ def matrix(self, references: List[SpectrumDocument], queries: List[SpectrumDocum
...
             reference_vectors[index_reference, 0:self.vector_size] = calc_vector(self.model,
                                                                                  reference,
                                                                                  self.intensity_weighting_power,
                                                                                  self.allowed_missing_percentage)
         n_cols = len(queries)
         query_vectors = numpy.empty((n_cols, self.vector_size), dtype="float")
-        for index_query, query in enumerate(queries):
+        for index_query, query in enumerate(tqdm(queries, disable=self.disable_progress_bar)):
```
Comment (sverhoeven) [inline spec2vec/Spec2Vec.py]:
```suggestion
        for index_query, query in enumerate(tqdm(queries, desc='Calculating vectors of query spectrums', disable=self.disable_progress_bar)):
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- Add optional progress bar for spec2vec.matrix() calculations (default is =False) [#43](https://github.com/iomega/spec2vec/pull/43)
```
Comment (sverhoeven) [inline CHANGELOG.md]: Double =/is
```suggestion
- Add optional progress bar for spec2vec.matrix() calculations (default is False) [#43](https://github.com/iomega/spec2vec/pull/43)
```

**Review by sverhoeven — CHANGES_REQUESTED** [review]
Seeing progress bar in tests

```shell
pytest --capture no -v tests/test_spec2vec.py 
...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 7294.44it/s]
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 8839.42it/s]
...
```

Good to know that setup-miniconda action moved.

Now if one is slow I dont know which set of spectrums it is working on. Please distinguish both progressbars with a description.

File: spec2vec/Spec2Vec.py (hunk 1 above)
Comment (florian-huber) [reply spec2vec/Spec2Vec.py]: That looks like a nice solution, thanks.

**Review by sverhoeven — APPROVED** [review]

**Comment by florian-huber** [comment]
Thanks a lot for reviewing Stefan!

**Pull request merged**